pkgs/applications/window-managers/hyprwm/hyprland-plugins/default.nix

```diff
@@ -1,27 +1,77 @@
-{ lib
-, callPackage
-, pkg-config
-, stdenv
-, hyprland
+{
+  lib,
+  callPackage,
+  fetchFromGitHub,
+  pkg-config,
+  stdenv,
+  hyprland,
 }:
 let
-  mkHyprlandPlugin = hyprland:
+  mkHyprlandPlugin =
+    hyprland:
     args@{ pluginName, ... }:
-    stdenv.mkDerivation (args // {
-      pname = "${pluginName}";
-      nativeBuildInputs = [ pkg-config ] ++ args.nativeBuildInputs or [ ];
-      buildInputs = [ hyprland ]
-        ++ hyprland.buildInputs
-        ++ (args.buildInputs or [ ]);
-      meta = args.meta // {
-        description = args.meta.description or "";
-        longDescription = (args.meta.longDescription or "") +
-          "\n\nPlugins can be installed via a plugin entry in the Hyprland NixOS or Home Manager options.";
+    stdenv.mkDerivation (
+      args
+      // {
+        pname = "${pluginName}";
+        nativeBuildInputs = [ pkg-config ] ++ args.nativeBuildInputs or [ ];
+        buildInputs = [ hyprland ] ++ hyprland.buildInputs ++ (args.buildInputs or [ ]);
+        meta = args.meta // {
+          description = args.meta.description or "";
+          longDescription =
+            (args.meta.longDescription or "")
+            + "\n\nPlugins can be installed via a plugin entry in the Hyprland NixOS or Home Manager options.";
+        };
+      }
+    );
+
+  # shared src for upstream hyprland-plugins repo
+  hyprland-plugins-src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = "hyprland-plugins";
+    rev = "d92d9d6cd93b731143119cbcab54dcc7a0481131";
+    hash = "sha256-Bw3JRBUZg2kmDwxa/UHvD//gGcNjbftTj2MSeLvx1q8=";
+  };
+
+  # function generating derivations for all plugins in hyprland-plugins
+  hyprland-plugins =
+    lib.mapAttrs
+      (
+        name: description:
+        (
+          { cmake, hyprland }:
+          mkHyprlandPlugin hyprland {
+            pluginName = name;
+            inherit (hyprland) version;
+
+            src = "${hyprland-plugins-src}/${name}";
+            nativeBuildInputs = [ cmake ];
+            meta = {
+              homepage = "https://github.com/hyprwm/hyprland-plugins";
+              description = "Hyprland ${description} plugin";
+              license = lib.licenses.bsd3;
+              maintainers = with lib.maintainers; [ fufexan ];
+              platforms = lib.platforms.linux;
+            };
+          }
+        )
+      )
+      {
+        borders-plus-plus = "multiple borders";
+        csgo-vulkan-fix = "CS:GO/CS2 Vulkan fix";
+        hyprbars = "window title";
+        hyprexpo = "workspaces overview";
+        hyprtrails = "smooth trails behind moving windows";
+        hyprwinwrap = "xwinwrap-like";
       };
-    });
 
-  plugins = {
-    hy3 = { fetchFromGitHub, cmake, hyprland }:
+  plugins = hyprland-plugins // {
+    hy3 =
```

Comment:
Can hy3 be moved to another file? and then imported here?

Reply:
Sure. I'll leave `default.nix` as the entrypoint.